src/net_processing.cpp

```diff
@@ -2535,6 +2541,11 @@ void PeerManager::ProcessMessage(CNode& pfrom, const std::string& msg_type, CDat
         return;
     }
 
+    if (msg_type == NetMsgType::SENDADDRV2) {
```

Comment:
I think we want to disconnect here if the sendaddrv2 message is received after VERACK (assuming that the suggested change to the BIP here: https://github.com/bitcoin/bips/pull/1043#discussion_r537711662 is taken)

Reply:
In theory we are free to not strictly enforce the bip. So even with the suggested change, the code here as is shouldn't cause any issues.

Reply:
Disconnecting if we receive `sendaddrv2` after `verack` might make sense for the following reasons:

* Allowing `sendaddrv2` after `verack` defeats the purpose of knowing for sure that the peer does not support `addrv2` (if we don't receive `sendaddrv2` before `verack`)
* If an odd implementation starts sending `sendaddrv2` after `verack` and we accept it, then we would be forever doomed to accept it because if we start disconnecting at some point in the future, then we would "break the existent app and network" (sounds familiar?). 
* It is what we do wrt wtxidrelay

Reply:
Agreed. I'd be in favor of aligning BIPs 155 and 339 on this, and their implementations, unless there is a good reason not to.

Reply:
closing as resolved

Reply:
See this comment which provides reasons not to disconnect: https://github.com/bitcoin/bitcoin/pull/20646#issuecomment-744704209